### PR TITLE
Validates the block before signing in Sumeragi

### PR DESCRIPTION
**Pull request by eadventurous**
Signed-off-by: Egor Ivkov <e.o.ivkov@gmail.com>

### CODE REVIEW

File: iroha/src/sumeragi.rs [hunk 1]
```diff
@@ -426,31 +426,42 @@ impl Sumeragi {
...
-                //TODO: send to set b so they can observe
             }
             Role::ProxyTail => {
                 if self.voting_block.write().await.is_none() {
-                    *self.voting_block.write().await = Some(VotingBlock::new(block));
+                    let mut block = block.clone();
+                    let verified_signatures = block.verified_signatures();
+                    block.signatures.clear();
+                    block.signatures.append(&verified_signatures);
+                    *self.voting_block.write().await = Some(VotingBlock::new(block))
```
Comment (humb1t) [inline iroha/src/sumeragi.rs:464]: Would be great to place it in some method with a good name. Like `VotingBlock::from(block.verify_signatures());`

File: iroha/src/block.rs [hunk 2]
```diff
@@ -163,6 +163,14 @@ impl SignedBlock {
     pub fn verified_signatures(&self) -> Vec<Signature> {
         self.signatures.verified(&self.hash())
     }
+
+    /// Returns `SignedBlock` with verified signatures.
+    pub fn verify_signatures(mut self) -> SignedBlock {
```
Comment (humb1t) [inline iroha/src/block.rs:168]: Maybe add new Block type `VerifiedBlock`? O make it impossible to have unverified signatures in `SignedBlock`?

Comment (eadventurous) [reply iroha/src/block.rs]: Actually I missed the fact that it is not needed to verify signatures in the places that we discussed. They are verified later along with the check of Roles at commit stage. So I removed this method.

Comment (eadventurous) [reply iroha/src/block.rs]: In fact we have already discussed it in another PR :smile:

**Pull request merged**